packages/os/cfsignal.service

```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Send signal to CloudFormation Stack
+After=network-online.target settings-applier.service
+Requires=settings-applier.service
+Wants=network-online.target
+
+[Service]
+Type=simple
+RemainAfterExit=true
+ExecStart=/bin/cfsignal
+
+[Install]
+RequiredBy=preconfigured.target
```

Comment:
This service should run almost at the very end of the boot sequence, right? If so, it should be `WantedBy=multi-user.target`. I think the weak dependency (`WantedBy`) should be sufficient since I don't think the `multi-user.target` should fail if this service fails.

Also could you please add `RefuseManualStart=true` and `RefuseManualStop=true`, I think it makes sense to set them  up, since as  you described `simple` is  a `oneshot`,  and we try to use these settings in all our `oneshot` services.

Reply:
no, the service must be run as soon as network is available and bottlerocket write the configuration files [settings-applier.service] after having read the user-data (we need them for stack-name and logical-resource-id).
This way we can intercept more boot fail causes and signal in advance cloudformation stack that ASG creation failed (without need to wait for the timeout)

Cfsignal invoke `systemctl is-system-running` command using `--wait` option, this is the key to know if boot is successful or not.

Looking at other services unit files i see that when `RefuseManualStart` is used there is a comment like:
```
# Block manual interactions with this service, since it could leave the system in an
# unexpected state
```
i do not think that reinvoking cfsignal could cause a problem.

Reply:
It needs to run early in the boot sequence, so that if we fail to transition to `configured.target` or `multi-user.target`, it can detect that failure and send the appropriate signal.

Reply:
`RefuseManualStart=true` and `RefuseManualStop=true` shouldn't be needed with the change to `WantedBy` that came in after your comment 😁